File: frontend/mobile/lib/services/user/me.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';
import 'package:area_mobile/types/user/me.dart';

Future<ServiceReturn<UserMe>> getMe() async {
  try {
    final response = await dio.get<Map<String, dynamic>>('/me');

    final responseData = response.data;
    if (responseData != null) {
      return ServiceReturn(data: UserMe.fromJson(responseData));
    } else {
      return const ServiceReturn(error: "API response data is null");
    }
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}

Future<ServiceReturn<void>> updateProfile(String language, String theme,
    {String? email}) async {
  try {
    var data = {"language": language, "theme": theme};
    if (email != null) data["email"] = email;

    await dio.patch("/me", data: data);
    return const ServiceReturn(data: null);
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}

File: frontend/web/services/user/me.ts
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";
import { ServiceReturn } from "@/types/api";
import { User, UserProfileUpdate } from "@/types/user";

const getProfile = async (): Promise<ServiceReturn<User>> => {
	try {
		const response = await axiosInstance.get<User>(`/me`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

const updateProfile = async (data: UserProfileUpdate): Promise<ServiceReturn<void>> => {
	try {
		const response = await axiosInstance.patch(`/me`, data);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export type UserStats = {
	workflowRuns: number;
	workflowErrors: number;
	workflows: number;
	activeWorkflows: number;
};
const getStats = async (): Promise<ServiceReturn<UserStats>> => {
	try {
		const response = await axiosInstance.get<UserStats>(`/workflows/summary`);
		return { data: response.data, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export { getProfile, updateProfile, getStats };
